api/app/Models/Pool.php

```diff
@@ -113,12 +114,48 @@ public function publishedPoolCandidates(): HasMany
 
     public function essentialSkills(): BelongsToMany
     {
-        return $this->belongsToMany(Skill::class, 'pools_essential_skills')->withTrashed(); // always fetch all skills for a pool
+        return $this->belongsToMany(Skill::class, 'pools_skills')
+            ->withTrashed() // pool-skills always fetches soft-deleted skill models
```

Comment:
I don't understand why this relationship needs to always return trashed skills?  I know we have to deal with this in #7833 but I'm not sure we want to make this change in here.

Reply:
Its because if a Pool has been soft-deleted (ie deprecated) but was already added to a pool in the past, we still want to return it

Reply:
There is no change though, it was `withTrashed()` before and after, spacing and comments is all that changed

Reply:
Oh man, I missed that too.  Lol, I need to pay more attention in my reviews.  :laughing: